src/ILCompiler.Compiler/src/IL/Stubs/Bootstrap/BootstrapMain.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Internal.TypeSystem;
+using Internal.IL;
+using Internal.IL.Stubs;
+using ILCompiler.Compiler.IL.Stubs;
+using System;
+
+namespace Internal.IL.Stubs.Bootstrap
+{
+    public sealed class BootstrapMainMethod : BootstrapMethod
+    {
+        private BootstrapData _data;
+        private ILEmitter _emitter;
+        private ILCodeStream _codeStream;
+
+        public BootstrapMainMethod(BootstrapData data)
+            : base(data.OwningType)
+        {
+            _data = data;
+        }
+
+        public override string Name
+        {
+            get
+            {
+                return "__Internal_IL_Stubs_BootstrapMain";
+            }
+        }
+
+        public override MethodIL EmitIL()
+        {
+            _emitter = new ILEmitter();
+            _codeStream = _emitter.NewCodeStream();
+
+            TypeDesc type = Context.SystemModule.GetType("System.Runtime", "BootstrapHelpers");
```

Comment:
The IL for this method should be (in pseudo code):

```
call BoostrapHelpers.Initialize

string[] args = BoostrapHelpers.GetCommandLineArgs(argc, argv);

// This needs to handle the different signatures for main, e.g. it can return void or int, or can tak no arguments or string[].
int exitcode = actual main(args)

call BoostrapHelpers.Shutdown

return exit code
```